internal/middleware/correlation_id.go

```diff
@@ -0,0 +1,28 @@
+package middleware
+
+import (
+	"net/http"
+
+	"github.com/RHEnVision/provisioning-backend/internal/ctxval"
+)
+
+func CorrelationID(next http.Handler) http.Handler {
+	fn := func(w http.ResponseWriter, r *http.Request) {
+		ctx := r.Context()
+
+		// Edge request id
+		corrId := r.Header.Get("Provisioning-Wizard-Id")
+		if corrId != "" {
+			ctx = ctxval.WithCorrelationId(ctx, corrId)
+		}
+
+		// Store in response headers for easier debugging
+		w.Header().Set("Provisioning-Wizard-Id", corrId)
+		logger := ctxval.Logger(ctx).With().Str("correlation_id", corrId).Logger()
+		logger.Trace().Msgf("Added correlation id %s to logger", corrId)
+
+		ctx = ctxval.WithLogger(ctx, &logger)
+		next.ServeHTTP(w, r.WithContext(ctx))
+	}
+	return http.HandlerFunc(fn)
+}
```

Comment:
The middleware and context is named "correlation", in the comment you have "edge request" and the header is named "provisioning wizard id".

I think we should be consistent, just a tad :-D

I don’t have a preference, just pick one term and use it everywhere.

Also note that custom HTTP headers should start with "X-" prefix as that namespace is reserved for custom headers. So for example if you choose this feature to be "correlation_id", then the header should be named `X-Correlation-Id`.